xgboost_ray/matrix.py

```diff
@@ -306,7 +343,8 @@ def load_data(self,
 
         if self.label is not None and not isinstance(self.label, str) and \
            not (isinstance(self.data, pd.DataFrame) and
-                isinstance(self.label, pd.Series)):
+                isinstance(self.label, pd.Series)) and \
+           not (_is_modin_df(self.data) and _is_modin_series(self.label)):
```

Comment:
Maybe move this condition logic to a separate function? Something like `is_valid_label`?

Reply:
Since we only need this once at this place I'll leave it here for the time being. If this grows another time, we should move this